src/request/swap-iframe/SwapIFrameApi.js

```diff
@@ -0,0 +1,385 @@
+
+        if (!storedData) throw new Error('No swap stored in SessionStorage');
+
+        /** @type {{keys: {nim: string, btc: string[]}, request: any}} */
+        const { keys: privateKeys, request: storedRawRequest } = (JSON.parse(storedData));
+
+        if (request.fund.type === 'NIM' || request.redeem.type === 'NIM') {
+            if (!privateKeys.nim) throw new Error('No NIM key stored in SessionStorage');
+            if (privateKeys.nim.length !== 64) throw new Error('Invalid NIM key stored in SessionStorage');
+        }
+
+        if (request.fund.type === 'BTC' || request.redeem.type === 'BTC') {
+            if (!privateKeys.btc) throw new Error('No BTC key list stored in SessionStorage');
+            if (!privateKeys.btc.length) throw new Error('No BTC keys stored in SessionStorage');
+            if (privateKeys.btc.some(key => !key)) throw new Error('Empty BTC key stored in SessionStorage');
+            if (privateKeys.btc.some(key => key.length !== 64)) {
+                throw new Error('Invalid BTC key stored in SessionStorage');
+            }
+        }
+
+        // Deserialize stored request
+        /** @type {Parsed<KeyguardRequest.SignSwapRequest>} */
+        const storedRequest = {
+            ...storedRawRequest,
+            ...(storedRawRequest.fund.type === 'NIM'
+                ? {
+                    fund: {
+                        ...storedRawRequest.fund,
+                        transaction: Nimiq.Transaction.fromPlain(storedRawRequest.fund.transaction),
+                    },
+                } : {}
+            ),
+            ...(storedRawRequest.redeem.type === 'NIM'
+                ? {
+                    redeem: {
+                        ...storedRawRequest.redeem,
+                        transaction: Nimiq.Transaction.fromPlain(storedRawRequest.redeem.transaction),
+                    },
+                } : {}
+            ),
+            ...(storedRawRequest.fund.type === 'BTC'
+                ? {
+                    fund: {
+                        ...storedRawRequest.fund,
+                        inputs: storedRawRequest.fund.inputs.map(
+                            /**
+                             * @param {any} input
+                             * @returns {ParsedBitcoinTransactionInput}
+                             */
+                            input => ({
+                                ...input,
+                                witnessUtxo: {
+                                    ...input.witnessUtxo,
+                                    script: BitcoinJS.Buffer.from(input.witnessUtxo.script, 'hex'),
+                                },
+                            }),
+                        ),
+                    },
+                } : {}
+            ),
+        };
+
+        // Parse request
+        /** @type {Parsed<KeyguardRequest.SignSwapTransactionsRequest>} */
+        // @ts-ignore Missing the following properties: fund, redeem
```

Comment:
I am not too fond of this `@ts-ignore`.
Maybe assemble the parts first and the `parsedRequest` last?